patchAPI/h/Point.h

```diff
@@ -157,6 +157,11 @@ Location(PatchFunction *f, PatchBlock *b, Dyninst::Address a, InstructionAPI::In
 // Used in PointType definition
 #define type_val(seq) (0x00000001u << seq)
 
+enum SnippetType {
```

Comment:
The C++ standard doesn't require enumerators to be ordered. Since they are being used for sorting, they should get explicit values. Ideally, this would be a [scoped enum](https://en.cppreference.com/w/cpp/language/enum.html#Scoped_enumerations), but that would be a pain to use for sorting. The next best thing is to put an anonymous enum in a namespace. The enumerators would then be referenced using a scoped name instead of a bareword. For example, `SnippetType::PROLOGUE` instead of `PROLOGUE`.

```suggestion
namespace SnippetType {
  enum {
    PROLOGUE = 0,
    REGULAR = 1,
    EPILOGUE = 2
  };
}
```

I would also add a comment saying that the order of these values is important since it determines the ordering of the snippets. That way, the next person will know.

Reply:
It does require them be order in increasing order, unless you explicitly define an enumerator's value. WIthout explicit definition the first enumerator is always `0`, and other enumerators without an explicit definition have the value `<PREVIOUS_ENUMERATOR> + 1`.

I don't think it would be a pain for sorting if you used a scoped enum `enum class SnippetType` (without explicit values). Comparison of a scoped enum with another is allowed (arithmetic and conversions are not).